workflow/controller/operator.go

```diff
@@ -440,12 +440,13 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 	}
 
 	var onExitNode *wfv1.NodeStatus
-	if woc.execWf.Spec.HasExitHook() && woc.GetShutdownStrategy().ShouldExecute(true) {
+	if woc.execWf.Spec.HasExitHook() {
```

Comment:
Yeap indeed. The other approach I thought of is to use a similar `woc.failSuspendedAndPendingNodesAfterDeadlineOrShutdown()` which iterates the nodes and updates the phase only if there are on running and shutdown terminate. 

However, I also need to fetch the onExit node cause with the current state even if I terminate the workflow. Instead of getting the failed status on the workflow we get successful workflow status cause we never evaluate the `onExit` node (due to the `if`)

Reply:
Or I could have a look to amend the above statuses, to report the correct `Skipped` status for `stepGroup` and `dag`

Reply:
@tczhao I update the logic with the if `woc.GetShutdownStrategy().ShouldExecute(true)`. The change should check if the `oneExitNode` exists then proceed with the `executeTemplate` func. Otherwise, proceed only if `woc.GetShutdownStrategy().ShouldExecute(true)` is true.

That way we avoid the execution of the `exitNode` when we terminate (tested both of the above cases the `exitNode` was `nil`).